frontend/components/exercise/FillInTheBlank.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState } from "react";
+
+interface FillInTheBlankProps {
+  id: string;
+  answer: string;
+  correctAnswer: string;
+  onAnswerChange: (answer: string) => void;
+}
+
+const FillInTheBlank: React.FC<FillInTheBlankProps> = ({
+  id,
+  answer,
+  correctAnswer,
+  onAnswerChange,
+}) => {
+  const [isActive, setIsActive] = useState(false);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    onAnswerChange(e.target.value);
+  };
+
+  const handleReset = () => {
+    onAnswerChange("");
+  };
+
+  const containerWidth =
+    Math.max(correctAnswer.length, "Type here".length) * 10 + 20;
+
+  return (
+    <div
+      className="relative inline-block flex h-[32px] items-center rounded-md border border-[#EDEDED] bg-white"
+      style={{
+        width: containerWidth,
+        padding: "0px 10px 0px 12px",
+      }}
+    >
+      <input
+        type="text"
+        value={answer}
+        onChange={handleInputChange}
+        onFocus={() => setIsActive(true)}
+        onBlur={() => setIsActive(false)}
+        className={`h-full w-full font-medium outline-none text-sm ${
+          isActive
+            ? "text-[#BFBFBF] placeholder-[#BFBFBF]"
```

Comment:
overall comment: can you fix the colors please isn't it easier to use the figma colors since they have names anyway ?